src/NUnitFramework/framework/Internal/Commands/ConstructFixtureCommand.cs

```diff
@@ -37,20 +38,40 @@ public class ConstructFixtureCommand : BeforeTestCommand
         public ConstructFixtureCommand(TestCommand innerCommand)
             : base(innerCommand)
         {
-            Guard.ArgumentValid(Test is TestSuite, "ConstructFixtureCommand must reference a TestSuite", nameof(innerCommand));
```

Comment:
This line is what triggered my comments above... ConstructFixtureCommand was designed to work on suites, represented by a `CompositeWorkItem`. That's in part reflected in the fact that it's a `BeforeTestCommand` rather than a `BeforeAndAfterTestCommand`. Repurposing it to work on both suites and test cases introduces a bit more complexity.

I recognize that it's a bit hard to see that we have two different sorts of commands, since they all work against `Tests` and dynamically determine if the test is a suite or not. That's because of the history... we first tried to make one type of command work in both environments, but it turned out that `CompositeWorkItems` needed two chains of commands, whereas `SimpleWorkItems` only need one. It was my hope to come back and refactor this at some point, but it never happened. :disappointed: